homeassistant/components/scrape/sensor.py

```diff
@@ -169,7 +169,8 @@ def __init__(
 
     def _extract_value(self) -> Any:
         """Parse the html extraction in the executor."""
-        raw_data = self.coordinator.data
+        if (raw_data := self.coordinator.data) is None:
+            return None
```

Comment:
This allows the sensor to be created even if the coordinator does not (yet) have data.
Should we perhaps not create the sensor if coordinator.data is empty?

Reply:
We do have a block for platform yaml
https://github.com/home-assistant/core/pull/81259/files#diff-b6ea9b97ed0d715cd77e8234539c7ad23900bb4036b9b63d24ff3d003577b20eR101-R102

Reply:
Moved `PlatformNotReady` to not load platform if data not available